vcd/resource_vcd_vapp_vm.go

```diff
@@ -329,7 +329,7 @@ func vmSchemaFunc(vmType typeOfVm) map[string]*schema.Schema {
 						Description:  "Network type to use: 'vapp', 'org' or 'none'. Use 'vapp' for vApp network, 'org' to attach Org VDC network. 'none' for empty NIC.",
 					},
 					"ip_allocation_mode": {
-						Optional:     true,
+						Required:     true,
```

Comment:
Mainly one query if you could check if there is a test of a VM that does not have `network` declared at all and it passes.

I think part of the reason that all fields where optional is that if one had a `Required` field within nested list - it would automatically force the root component `network` to be required. This was especially true when this resource had (long time ago)only two plain fields to specify a `network_name` and `ip`. The new `network` block had to coexist until we released 3.0 and removed the deprecated fields.

Reply:
Good point, will run all `vm` tests and get back to this comment